clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -1056,6 +1133,15 @@ void abortableError(RuntimeException e) {
             transitionToAbortableError(e);
         }
 
+        void recoverableError(RuntimeException e) {
```

Comment:
The name "recoverable" is a bit misleading since the function may result in a fatal error. I'm struggling to think of a better name though. Something like `aborableErrorIfPossible`?